### DISCUSSION TITLE
Can Contours of PlotMapView be esported to Shapefiles?

### DISCUSSION THREAD

**Discussion by javgs-bd** [modflowpy/flopy#1527] [Q&A] [answered]
I'm trying to export the contours of a PlotMapView to a shapefile, and the only way I'm identifying is by building geometries from the array of points contained in the matplotlib collections. I wonder if there is an easier way that I am missing. 

So, looking at this notebook

https://github.com/modflowpy/flopy/blob/develop/examples/Notebooks/flopy3.3_PlotMapView.ipynb

and this image (plotmapview) coming from notebook cell 31:

![image](https://user-images.githubusercontent.com/74055695/187738168-f517dfbb-2ccc-47ef-9f46-7ba2ea393473.png)

Is there an already flopy built in command that I am missing and that will allow me to send the lines in the contour_set to a shapefile?

**Comment by javgs-bd** [comment]
Just to complement, I can get an array of points using ````contour_set.collections[0].get_paths()[i]```` where i is the index of one of the lines. But it is cumbersome to know which line I'm getting and programing the whole loop to build Shapely geometries and then export. I'm just wondering if I'm missing a more direct way.

**Comment by wpbonelli** [comment]
There is a [function](https://github.com/modflowpy/flopy/blob/develop/flopy/plot/plotutil.py#L2071) to go in the opposite direction (shapefile to patch collection) in the plot utilities ~~but I don't believe there's a way to export patch collections straight to shape file. May be worth adding however.~~

**Comment by christianlangevin** [comment, marked as answer]
I believe there is a function called export_contours() in flopy.export.utils.  Pretty sure that should help.

**Reply by wpbonelli** [reply]
@langevin-usgs my mistake

**Reply by javgs-bd** [reply]
Ahh it sure did!
Many thanks. You guys rock.

For whomever might be looking for the same:
https://flopy.readthedocs.io/en/latest/source/flopy.export.utils.html?highlight=flopy.export.utils#flopy.export.utils.export_contours

Following the cited notebook I believe the line to use there would have been:

````flopy.export.utils.export_contours(vertex_ml6.modelgrid, 'output.shp', contour_set, fieldname='head', epsg=None, prj=None)````

**Reply by wpbonelli** [reply]
#1528 adds test coverage for `export_contours` and mentions it in the map view plot example notebook

may also be worth either adding a `modelgrid` param to `export_contourf` or removing it from `export_contours` for consistency's sake? PR opts for latter right now but happy to update.
